ui/components/app/detected-token/detected-token.js

```diff
@@ -108,17 +177,53 @@ const DetectedToken = ({ setShowDetectedTokens }) => {
         asset_type: AssetType.token,
       },
     });
-    const deSelectedTokensAddresses = deSelectedTokens.map(
-      ({ address }) => address,
-    );
-    await dispatch(
-      ignoreTokens({
-        tokensToIgnore: deSelectedTokensAddresses,
-        dontShowLoadingIndicator: true,
-      }),
-    );
-    setShowDetectedTokens(false);
-    setPartiallyIgnoreDetectedTokens(false);
+
+    if (process.env.PORTFOLIO_VIEW) {
+      // group deselected tokens by chainId
+      const groupedByChainId = deSelectedTokens.reduce((acc, token) => {
+        const { chainId } = token;
+        if (!acc[chainId]) {
+          acc[chainId] = [];
+        }
+        acc[chainId].push(token);
+        return acc;
+      }, {});
+
+      const promises = Object.entries(groupedByChainId).map(
+        async ([chainId, tokens]) => {
+          const chainConfig = configuration[chainId];
+          const { defaultRpcEndpointIndex } = chainConfig;
+          const { networkClientId: networkInstanceId } =
+            chainConfig.rpcEndpoints[defaultRpcEndpointIndex];
+
+          await dispatch(
+            ignoreTokens({
+              tokensToIgnore: tokens,
+              dontShowLoadingIndicator: true,
+              networkClientId: networkInstanceId,
```

Comment:
In the future we can make `ignoreTokens` and other controller APIs just accept chain ids instead of network client ids